byway/src/main/webapp/script.js

```diff
@@ -15,14 +15,144 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+window.onload = function() {
+  setProgressBar(1);
+  getStartDestination();
+  getLocations();
+};
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function setProgressBar(pageNumber){
+  var ul = document.getElementById("progressbar");
+  var items = ul.getElementsByTagName("li");
+  items[pageNumber-1].className = 'active';
+}
+
+var defaultCenter = {
+  lat: 40.712776,
+  lng:-74.005974
+};
+var userlatlng = {lat:null , lng: null}
+
+function initAutocomplete() {   
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: defaultCenter.lat, lng: defaultCenter.lng },
+    zoom: 13,
+    mapTypeId: "roadmap"
+  });
+
+
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      userlatlng = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+      map.setCenter(userlatlng);
+    }, function() {
+         map.setCenter(defaultCenter);
+        });
+  } 
+  else {
+    window.alert("Geolocation failed");
+  }
+  // Create the search boxes and link them to the UI elements.
+  createSearchBox(map,'pac-input');
+  createSearchBox(map,'pac-input-2');
+  
+}
+
+function createSearchBox(map,container){
```

Comment:
Since javascript doesn't have typing, it would be useful to document what type of parameters you function expects. You can follow the style guide recommendations outlined here:
https://google.github.io/styleguide/jsguide.html#features-functions-top-level-functions

e.g. you could update this description like such:
```
/*
 * Creates a search box
 * @param {google.maps.Map} map
 * @param {string} container
 */
```

Reply:
done